src/main/scala/org/bitcoins/core/script/crypto/HashType.scala

```diff
@@ -1,47 +1,145 @@
 package org.bitcoins.core.script.crypto
 
-import org.bitcoins.core.util.{BitcoinSUtil}
+import org.bitcoins.core.number.Int32
+import org.bitcoins.core.script.crypto.SIGHASH_ALL.SIGHASH_ALLImpl
+import org.bitcoins.core.util.{Factory, BitcoinSUtil}
 
 /**
  * Created by chris on 1/18/16.
  */
 sealed trait HashType {
-  def byte : Byte
-  def hex : String = BitcoinSUtil.encodeHex(byte)
+  type HashType
+  def num : Int32
 }
 
+trait HashTypeOperations extends Factory[HashType] {
```

Comment:
This can be just be `object HashType extends Factory[HashType] { ... }` instead of being a trait